### DISCUSSION TITLE
Is there a way to change default font to be reverse-type?

### DISCUSSION THREAD

**Discussion by t5404tmz** [todotxt/todo.txt-cli#329] [Q&A] [answered]
**Do you want to request a *feature* or report a *bug*?**
Request a feature.

**What is the current behavior?**
Default font is not reverse-type.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
Reverse type font is easier to read.  I changed the code for priority records to display reverse type fonts but I can't figure out how to change the default, non-priority type entries.

**Which versions todo.sh are you using?**
> Run `todo.sh -V`
TODO.TXT Command Line Interface v@DEV_VERSION@

Homepage: http://todotxt.org
Code repository: https://github.com/todotxt/todo.txt-cli/
Contributors: https://github.com/todotxt/todo.txt-cli/graphs/contributors
License: https://github.com/todotxt/todo.txt-cli/blob/master/LICENSE

**Which Operating System are you using?**
Ubuntu 16.04

**Which version of bash are you using?**
> Run `bash --version`
GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

**Comment by inkarkat** [comment, marked as answer]
_(I assume that by _reverse-type font_ you mean inverting foreground and background colors; e.g. turning black-on-white to white-on-black.)_

Like most command-line applications, todo.sh uses the default terminal configuration of colors and font attributes, and optionally offers highlightings for special elements (like prioritized tasks, dates, metadata). If "Reverse type font is easier to read" for you, why don't you simply reconfigure your terminal that way (and all command-line use and applications would benefit from it)?!

---

That said, there is a way to reverse the entire todo.sh output, via the `TODOTXT_FINAL_FILTER`. Put this into your `todo.cfg` config file:

    export TODOTXT_FINAL_FILTER='printf ^[[07m; cat; printf ^[[0m'
    export DEFAULT='\\033[07m'

The final filter is applied last, so if you switch the terminal to reverse output (via the `^[[07m` ANSI escape; note that `^[` is a single literal Escape character, usually typed via Ctrl+V followed by Escape), and undo that at the end, you should get the desired result. Any terminal resets inside the todo.sh output have to be changed to keep the reverse attribute, so `DEFAULT` is adapted as well. (Instead of simply reverting, any foreground / background color modification could be done this way.)

I'd still see this more as a proof of concept than a recommendation to use; reconfiguring the terminal would be the way to go for me. Please re-open the issue if I misunderstood any of this.
